src/sshkeys/sshkeys.controller.ts

```diff
@@ -0,0 +1,66 @@
+import { Body, Controller, Delete, Get, HttpException, HttpStatus, Param, Post } from '@nestjs/common';
+import { SshKeysService } from './sshkeys.service';
+import { AdminOnly } from 'src/auth/decorators/adminonly.decorator';
+import { CreateSshKeyDto } from './dto/create-sshkey.dto';
+import { GetUser } from 'src/auth/decorators/user.decorator';
+import { UserDecoratorType } from 'src/auth/types/user-decorator.type';
+
+@Controller('sshkeys')
+export class SshKeysController {
+    constructor(
+        private sshkeysService: SshKeysService
+    ) {}
+
+    // POST /sshkeys/my
+    // This action adds a key ${setSshDto} to a user's keys
+    @Post('my')
+    async createSshKey(@Body() createSshDto: CreateSshKeyDto, @GetUser() user: UserDecoratorType) {
+        try {
+            if (!await this.sshkeysService.createSshKey(user.sub, createSshDto)) {
+                return new HttpException("Unable to add this ssh key. Please verify the key and name.", HttpStatus.BAD_REQUEST);
+            }
+            return "SSH key was successfully created."
+        } catch(err) {
+            throw new HttpException("SSH key could not be added to this account.", HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    // DELETE /sshkeys/:username
+    // This action removes a key ${setSshDto} to a user's keys
+    @Delete('my/:uuid')
+    async deleteSshKey(@Param('uuid') uuidSshKey: string, @GetUser() user: UserDecoratorType) {
+        try {
+            if(!await this.sshkeysService.removeSshKey(user.sub, uuidSshKey)) {
+                throw new HttpException("User or key not found.", HttpStatus.BAD_REQUEST)
+            }
```

Comment:
When I delete a ssh key that does not exist it's catched by the `try/catch` and returns an internal error `SSH key could not be removed`.

Reply:
Good point, it should be fixed by now !